src/Utilities/StdConstexprHelpers.hpp

```diff
@@ -0,0 +1,116 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/*!
+ * \ingroup Utilities
+ * These functions are constexpr reimplementations of some standard library
+ * functions, designed to work with cpp17::array.
+ */
+
+#pragma once
+
+#include "Utilities/Gsl.hpp"
+
+namespace cpp2b {
+/*!
+ * \ingroup Utilities
+ * Reimplementation of std::accumulate that is constexpr;
+ * taken from the LLVM source at
+ * https://github.com/llvm-mirror/libcxx/blob/master/include/numeric
+ */
+template <class ForwardIterator, class T>
+constexpr void iota(ForwardIterator first, ForwardIterator last,
```

Comment:
I think all of these should have the same exception specification as the standard says they do, even though we say everything should be `noexcept`. The reason being if someone needs exception handling in an STL function, they should have it.

Reply:
OK. When I checked, swap was noexcept, but iter_swap, accumulate, iota, reverse, and next_permutation were not. I'll match that in the next commit